### DISCUSSION TITLE
Holes after ball pivoting

### DISCUSSION THREAD

**Discussion by wikyburns** [cnr-isti-vclab/PyMeshLab#80] [Q&A]
Hello,

I have a little issue and I need help to solve it.

Actually i'm using this .ply and then reconstructing with "ball pivoting" with the default and when I export it to .obj it has a lot of holes on the plantar foot:

![image](https://user-images.githubusercontent.com/11525781/111833019-d52e2880-88f1-11eb-82b1-d854ddc51d6c.png)

How I can solve it?

Other question, any idea to how to "auto" remove all the not important mesh, instead the plantar foot?

Thanks

I attach the .ply and .obj already exported.

[scan_bug.zip](https://github.com/cnr-isti-vclab/PyMeshLab/files/6173623/scan_bug.zip)

**Comment by alemuntoni** [comment]
What do you mean with "not important mesh"? 
There is no such "semantic" description of meshes, if this is what you mean.
You could use the "select small disconnected components" and then the "remove selected vertices and faces" filters, or "remove isolated pieces" directly

**Reply by wikyburns** [reply]
Hi @alemuntoni,

And how about fixing the "bug" with the holes on the plantar foot? it only happens when i have a lot of points.

Regards,

Sebas

**Reply by wikyburns** [reply]
When I use the ball pivoting filter with the 100% of the points it generates holes on the plantar foot like the this image:
![image](https://user-images.githubusercontent.com/11525781/112328249-73393e80-8cb6-11eb-8fce-e801648f1977.png)

But when I remove the other points and leave only the plantar foot and then I apply the ball pivoting algorithm it work better:
![image](https://user-images.githubusercontent.com/11525781/112328469-a5e33700-8cb6-11eb-871f-a60eccea9dd6.png)


What I can do to obtain the same result as the second image without removing points?

Really thanks!

**Reply by wikyburns** [reply]
> What do you mean with "not important mesh"? 
> 
> There is no such "semantic" description of meshes, if this is what you mean.
> 
> You could use the "select small disconnected components" and then the "remove selected vertices and faces" filters, or "remove isolated pieces" directly

 Hi, any idea why is creating the holes?
